src/saiga/opengl/rendering/lighting/light_clusterer.h

```diff
@@ -285,5 +264,10 @@ class SAIGA_OPENGL_API Clusterer
     std::shared_ptr<BuildClusterComputeShader> buildClusterShader3D;
     std::shared_ptr<LightAssignmentComputeShader> buildLightToClusterMapShader;
 #endif
+    virtual bool fillImGui();
+
+   private:
+    void beginImGui(bool* p_open = NULL);
```

Comment:
why does this class need 3 imgui functions? Shouldn't one be enough?

Reply:
A clusterer has some "basic" ui elements, like "TileSize", this is in the light_clusterer imgui function. But some of the derived classes have some extra ui for example "lightDebug" in the cpu_plane_clusterer. I did not want to duplicate the code in the derived classes, but with one function, I would have to start an new ImGui-Window for each derived class. So I let the base clusterer begin the window. fill it with the basic stuff, the derived stuff and close the window again. :)